src/oracc_cmd.ts

```diff
@@ -0,0 +1,50 @@
+import * as child_process from 'child_process';
+import { homedir } from 'os';
+import * as path from 'path';
+import * as process from 'process';
+import { promisify } from 'util';
+import * as vscode from 'vscode';
+
+const execFile = promisify(child_process.execFile);
+
+// Thin wrapper around `execFile` which prepares the environment where we want to be running
+// the commands in.
+function run_cmd(file: string,
+                 args: string[] = [],
+                 options: Object = {
+                     cwd: homedir(),
+                     env: {
+                         ...process.env,
+                         ...{PATH: '/usr/local/oracc/bin:' + process.env.PATH},
+                     },
+                 }) {
+    return execFile(file, args, options);
+}
+
+// Run the `oracc` command with the given action.  This assumes the command is being run on
+// the server.
+export async function run_oracc(cmd: string[]) {
+    const editor = vscode.window.activeTextEditor;
+    const ext = path.parse(editor.document.uri.fsPath).ext;
+    if (ext !== ".glo") {
+        vscode.window.showWarningMessage('Oracc command can only be run from glossary files');
+        return;
+    }
+
+    try {
+        const { stdout } = await run_cmd('oracc', cmd);
+        vscode.window.showWarningMessage(`${stdout}`);
+    } catch (err) {
+        // If the `oracc` command cannot be spawned because it doesn't exist (and also if it
+        // isn't readable/executable from the user?), the error code is 'ENOENT'.  Instead,
+        // if the process itself throws a "no such file or directory" error, then `code ===
+        // 2`, which is different.  XXX: we could also use `os.hostname()` to check the
+        // machine where the command is being run, and decide what to do, if necessary.
```

Comment:
You mean deciding what error to show based on the hostname? I think the error code may be more reliable but not sure.

Reply:
I was thinking of allowing to run `oracc` only on the Oracc server and provide a clear error message to users if they aren't running it there, but that may be too restrictive, especially for local testing.

Reply:
Yeah, I think if users have the executable locally (some might?), we might as well let them try it. Unless it leads to confusion about whether the files are updated on the server or not - but that would be the same if they were doing it on the command line too.
I would say let's go with the simplest option for us :)